src/macports1.0/macports.tcl

```diff
@@ -1063,10 +1063,6 @@ match macports.conf.default."
         } else {
             set macports::universal_archs {i386 ppc}
         }
-    } elseif {[llength $macports::universal_archs] < 2} {
-        if {$os_major < 18} {
-            ui_warn "invalid universal_archs configured (should contain at least 2 archs)"
-        }
```

Comment:
We need some method of explaining to MacPorts when it should not offer universal variants. The proposal is that setting `universal_archs` to a single value (or no value) is that method. Do you have an alternate proposal?

Reply:
First of all, I agree that we should not rip out support for universal completely, as we might want to build for something like x86_64 and arm64 later.

When upgrading to Mojave, the default `macports.conf` will still contain `universal_archs x86_64 i386`, even after upgrading MacPorts. I am afraid that users will not know about that and simple continue to use their existing `macports.conf`. Therefore this is error-prone, if a single arch in `universal_archs` is supposed to be the indicator instead of "not supported on Mojave".

So if we go with "single arch in universal_archs", I would additionally propose to either silently filter `i386` from `universal_archs` as invalid on Mojave, or issue a warning to get users to update their `macports.conf`.

Reply:
> First of all, I agree that we should not rip out support for universal completely, as we might want to build for something like x86_64 and arm64 later.

Right. There have been rumors of future ARM-based Macs.

> When upgrading to Mojave, the default `macports.conf` will still contain `universal_archs x86_64 i386`, even after upgrading MacPorts.

Good point. Meanwhile, for new installations, it will contain only x86_64. That is confusing, and as I pointed out on the mailing list, at the latest when 10.15 is released we will need a strategy for how to handle that situation.

> I am afraid that users will not know about that and simple continue to use their existing `macports.conf`. Therefore this is error-prone, if a single arch in `universal_archs` is supposed to be the indicator instead of "not supported on Mojave".

A single arch (or no archs) in `universal_archs` is my currently proposed method of indicating that the universal variant should not be offered (on any OS version).

An alternative possibility is that, if it is not already exposed there, we expose `universal_supported` in macports.conf and make sure it's written to the default macports.conf file commented out. Then we can modify its default value either on 10.14 or 10.15 and everyone will get that new default when they upgrade.

> So if we go with "single arch in universal_archs", I would additionally propose to either silently filter `i386` from `universal_archs` as invalid on Mojave, or issue a warning to get users to update their `macports.conf`.

It is still my hope that 32-bit building can be supported after all on 10.14 using the 10.13 SDK as I [described on the mailing list](https://lists.macports.org/pipermail/macports-dev/2018-September/039395.html). So hopefully we don't introduce any changes to base that will complicate that.